plugins/main/public/components/common/search-bar/search-bar-service.ts

```diff
@@ -1,114 +1,148 @@
-    filters?: Filter[];
-    query?: any;
-    pagination?: {
-        pageIndex?: number;
-        pageSize?: number;
-    };
-    fields?: string[],
-    sorting?: {
-        columns: {
-            id: string;
-            direction: 'asc' | 'desc';
-        }[];
-    };
-    dateRange?: {
-        from: string;
-        to: string;
-    };
-}
+export type SearchParams = {
+  indexPattern: IndexPattern;
+} & tSearchParams;
 
-export const search = async (params: SearchParams): Promise<SearchResponse | void> => {
-    const { indexPattern, filters: defaultFilters = [], query, pagination, sorting, fields } = params;
-    if (!indexPattern) {
-        return;
-    }
-    const data = getPlugins().data;
-    const searchSource = await data.search.searchSource.create();
-    const fromField = (pagination?.pageIndex || 0) * (pagination?.pageSize || 100);
-    const sortOrder: OpenSearchQuerySortValue[] = sorting?.columns.map((column) => {
-        const sortDirection = column.direction === 'asc' ? 'asc' : 'desc';
-        return { [column?.id || '']: sortDirection } as OpenSearchQuerySortValue;
+export const search = async (
+  params: SearchParams,
+): Promise<SearchResponse | void> => {
+  const {
+    indexPattern,
+    filters: defaultFilters = [],
+    query,
+    pagination,
+    sorting,
+    fields,
+    aggs,
+  } = params;
+  if (!indexPattern) {
+    return;
+  }
+  const data = getPlugins().data;
+  const searchSource = await data.search.searchSource.create();
+  const fromField =
+    (pagination?.pageIndex || 0) * (pagination?.pageSize || 100);
+  const sortOrder: OpenSearchQuerySortValue[] =
+    sorting?.columns.map(column => {
+      const sortDirection = column.direction === 'asc' ? 'asc' : 'desc';
+      return { [column?.id || '']: sortDirection } as OpenSearchQuerySortValue;
     }) || [];
-    let filters = defaultFilters;
+  let filters = defaultFilters;
 
-    // check if dateRange is defined
-    if (params.dateRange && params.dateRange?.from && params.dateRange?.to) {
-        const { from, to } = params.dateRange;
-        filters = [
-            ...filters,
-            {
-                // @ts-ignore
-                range: {
-                    [indexPattern.timeFieldName || 'timestamp']: {
-                        gte: from,
-                        lte: to,
-                        format: 'strict_date_optional_time'
-                    }
-                }
-            }
-        ]
+  // check if dateRange is defined
+  if (params.dateRange && params.dateRange?.from && params.dateRange?.to) {
+    // when the from and to are equal, the to is now
+    /* ToDo: Search the OSD date range parser, to prevent this change and fix
+        when the fllter is "Today" and "This week". In those ranges the filter to and from are the same
+        Today: from: now/d, to: now/d
+        This Week: from: now/w to:now/w
+    */
+    if (params.dateRange.from === params.dateRange.to) {
+      params.dateRange.to = 'now';
```

Comment:
**nitpick**: can the user set the dateRange.from and dateRange.to to the same values, and they are not the cases for Today/This week. We should review how the Discover application is managing these cases.